{{cookiecutter.project_slug}}/requirements/requirements.in

```diff
@@ -1,3 +1,2 @@
-{%- if cookiecutter.sample_code != 'no' -%}
 click
-{% endif %}
\ No newline at end of file
+pytest
```

Comment:
There are good arguments for either way. Keeping the test environment close as close to the runtime environment as possible definitely seems sensible. The cleanest might be to add a `test-requirements.in` (and consequently a `test-environment.yml`), though whether adding additional files would be worth it over pragmatically adding `pytest` to `requirements.in` is questionable, at least as long it's only `pytest`...

For the actual runtime environment (`environment.yml`) it probably wouldn't make a difference in most cases, anyway, as chances are that some project in the dependency tree depends on `pytest` (as is, e.g., the case for [`pyflexplot`](https://github.com/MeteoSwiss-APN/pyflexplot/blob/master/environment.yml)).

I guess I would keep your change (i.e., `pytest` in `requirements.in`) and document the reasoning behind it, with a remark that in case of more complex testing setups that require multiple packages, adding `test-requirements.in` and `test-environment.yml` may be sensible.

Reply:
I would second your opinions here and go for the pragmatic solution, leave the test requirements (i.e. at the moment only pytest) in `requirements.in`. The created overhead in the runtime environment is likely marginal but it keeps the worklflow easier when we switch from pinned to unpinned environments.